int/hwsc-user-svc/proto/hwsc-user-svc.proto

```diff
@@ -0,0 +1,203 @@
+    // svc will check that if the email changed, go through users documents to check for duplicated email
+    // svc will then do partial or complete update on users document?
+    // svc returns OK || email is already taken error || service is down error
+
+    // if OK, chrome sets offical User object with copy User object, displays success msg to user
+    // else, chrome will display error msg
+    rpc ModifyUserInfo (UserRequest) returns (UserResponse) {}
+
+
+    // chrome stores email/password in an object and makes request to login from login page
+    // gateway passes email/password object
+    // svc will look through user document for matching email
+    // svc will then match password (dehash, then match)
+    // svc will start session????
+    // svc returns OK with User object (with password nulled?) and some kind of session token?
+    //      || email doesn't match error
+    //      || if email match, password doesn't match error
+    //      || service is down error
+
+    // if OK, chrome will store user object and whatever token?, redirect to search page
+    // else, chrome will display error msg
+    rpc AuthenticateUserLogin (UserRequest) returns (UserResponse) {}
+
+
+
+
+    // THINGS GET KIND OF CONFUSING FROM HERE --------------------------------------
+
+
+    // chrome makes request to add friend by providing email? or by first and/or last name?
+    // gateway passes email (or first and last name)
+    // svc will look through user documents for matching email or name
+    // svc returns OK with found Users || email/name not found error || service is down error
+
+    // if OK, chrome will display Users first and last name and ask for confirmation?
+    //      || display list of matched names and corresponding emails and ask user to select correct one
+    //   Once user selects or confirms, chrome will store the matched users object under friends object
+    // else, chrome will display not found msg or error msg
+    rpc LookUpUser (UserRequest) returns (UserResponse) {}
+
+
+    // chrome makes request to share documents with found user
+    // gateway passes
+    //    1) request makers User object
+    //    2) a list of DUIDS to share with
+    //    3) found User object
+    // svc will...
+    rpc ShareDocuments (UserRequest) returns (UserResponse) {}
+
+
+    // chrome makes request to delete document/s
+    // gateway passes
+    //      1) document/s to delete
+    //      2) User object
+    // svc will go through users object documents map and delete key's (duid)
+    // svc will...
+    rpc DeleteDocuments (UserRequest) returns (UserResponse) {}
+
+}
+
+
+
+
+message DocumentData {
+    bool public = 1;
+    repeated string shared_with = 2;
+}
+
+message User {
+    string uuid = 1;
+    string first_name = 2;
+    string last_name = 3;
+    string email = 4;
+    string password = 5;
+    string organization = 6;
+    map<string, DocumentData> documents = 7;
```

Comment:
Add a comment to track this field. name this ``user_documents``
// Key = duid in string
map<string, UserDocumentMetadata > user_documents = 7;

Reply:
Consider adding another field called:
```
// Key = friend's uuid
map<string, UserFriendMetadata> shared_to_me = 8;
```

Reply:
```
message UserFriendMetadata{
   // Key = friend's duid
  // Value = This can be anything. This helps O(1) lookup time for deletion and updating
  map<string, bool> shared_duid_to_me = 2;
}
```